pallets/storage-proofs/src/lib.rs

```diff
@@ -0,0 +1,493 @@
+    #[pallet::storage]
+    #[pallet::getter(fn challenges_queue)]
+    pub type ChallengesQueue<T: Config> =
+        StorageValue<_, BoundedVec<FileKeyFor<T>, ChallengesQueueLengthFor<T>>, ValueQuery>;
+
+    /// A priority queue of file keys that have been challenged manually.
+    ///
+    /// The difference between this and `ChallengesQueue` is that the challenges
+    /// in this queue are given priority over the others. So this queue should be
+    /// emptied before any of the challenges in the `ChallengesQueue` are dispatched.
+    /// This queue should not be accessible to the public.
+    /// The elements in this queue will be challenged in the coming blocks,
+    /// always ensuring that the maximum number of challenges per block is not exceeded.
+    /// A `BoundedVec` is used because the `parity_scale_codec::MaxEncodedLen` trait
+    /// is required, but using a `VecDeque` would be more efficient as this is a FIFO queue.
+    #[pallet::storage]
+    #[pallet::getter(fn priority_challenges_queue)]
+    pub type PriorityChallengesQueue<T: Config> =
+        StorageValue<_, BoundedVec<FileKeyFor<T>, ChallengesQueueLengthFor<T>>, ValueQuery>;
+
+    /// The block number of the last checkpoint challenge round.
+    ///
+    /// This is used to determine when to include the challenges from the `ChallengesQueue` and
+    /// `PriorityChallengesQueue` in the `BlockToChallenges` StorageMap. These checkpoint challenge
+    /// rounds have to be answered by ALL Storage Providers, and this is enforced by the
+    /// `submit_proof` extrinsic.
+    #[pallet::storage]
+    #[pallet::getter(fn last_checkpoint_block)]
+    pub type LastCheckpointBlock<T: Config> = StorageValue<_, BlockNumberFor<T>, ValueQuery>;
+
+    // Pallets use events to inform users when important changes are made.
+    // https://docs.substrate.io/v3/runtime/events-and-errors
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// A manual challenge was submitted.
+        /// [who, file_key_challenged]
+        NewChallenge(AccountIdFor<T>, FileKeyFor<T>),
+
+        /// A storage proof was rejected.
+        /// [storage_provider, proof, reason]
+        ProofRejected(AccountIdFor<T>, CompactProof, ProofRejectionReason),
+
+        /// A storage proof was accepted.
+        /// [storage_provider, proof]
+        ProofAccepted(AccountIdFor<T>, CompactProof),
+    }
+
+    // Errors inform users that something went wrong.
+    #[pallet::error]
+    pub enum Error<T> {
+        /// The ChallengesQueue is full. No more manual challenges can be made
+        /// until some of the challenges in the queue are dispatched.
+        ChallengesQueueOverflow,
+
+        /// The proof submitter is not a registered Storage Provider.
+        NotStorageProvider,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
```

Comment:
remove?

Reply:
Solved in [refactor: ⚰️ Remove unused #[pallet::hooks]](https://github.com/Moonsong-Labs/storage-hub/pull/8/commits/d31a41fa78d0b2a2879ffea1cf5a9878d5bc0e73)